Gems/Atom/RHI/DX12/Code/Source/RHI/CommandListBase.cpp

```diff
@@ -97,46 +125,123 @@ namespace AZ
             {
                 AZ_PROFILE_SCOPE(RHI, "CommandListBase: FlushBarriers");
 
-                m_commandList->ResourceBarrier((UINT)m_queuedBarriers.size(), m_queuedBarriers.data());
+                auto beginIt = m_queuedBarriers.begin();
+                decltype(beginIt) endIt;
+                AZStd::vector<D3D12_RESOURCE_BARRIER> barriers(m_queuedBarriers.size());
+                decltype(BarrierOp::m_cmdListState) currentState;
+                do
```

Comment:
Add a quick comment as to why we have this logic to  to set sample sample positions when flushing barriers in case it confuses someone.